lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexFunction.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.api;
+
+import java.io.Serializable;
+
+/**
+ * Typed user function.
+ * Implement your functions using this interface.
+ *
+ * @param <TInput> input type
+ * @param <TOutput> output type
+ */
+public interface VortexFunction<TInput extends Serializable, TOutput extends Serializable> extends Serializable {
```

Comment:
One way I can think of to avoid `java.io.Serializable` is to pre-load all user-defined functions at the time of Evaluator launch. That way Driver simply tells the class name of the function to Evaluator instead of serializing and sending the actual class. This also lets us avoid the redundant serialization/sending of same function classes.

I'm not sure how we can use Tang(pre-inject?) to make this happen and of the implications for the Vortex user api(need to let Vortex know all function classes the user will use before launching any tasks?) though.